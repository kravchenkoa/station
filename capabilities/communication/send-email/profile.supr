"""
Send Email
Send transactional email.
"""

name = "communication/send-email"
version = "2.0.0"

"""
Send Email
Send transactional email to one recipient.
Email can contain text and/or html representation.
"""
usecase SendEmail unsafe {
  input {
    """
    From
    The sender's email address.
    """
    from!


    """
    To
    The recipient's email address.
    """
    to!

    """
    Subject
    The subject of your email. See character length requirements according to RFC 2822.
    """
    subject!

    """
    Text
    The plain text email message.
    """
    text

    """
    HTML
    The HTML email message.
    """
    html

  }

  result {
    """
    Message Identifier
    The identifier is provider-specific and not unique.
    """
    messageId!
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title!

    """
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail
  }
}
