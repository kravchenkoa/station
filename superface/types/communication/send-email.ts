import { TypedProfile, typeHelper } from '@superfaceai/one-sdk';
/**
 * Send transactional email to one recipient
 * Email can contain text and/or html representation
 **/
export interface SendEmailInput {
  from: unknown;
  to: unknown;
  subject: unknown;
  text?: unknown;
  html?: unknown;
}
/**
 * Send transactional email to one recipient
 * Email can contain text and/or html representation
 **/
export interface SendEmailResult {
  messageId?: unknown;
}
const profile = {
  "SendEmail": typeHelper<SendEmailInput, SendEmailResult>()
};
export type CommunicationSendEmailProfile = TypedProfile<typeof profile>;
export const communicationSendEmail = {
  "communication/send-email": profile
};
