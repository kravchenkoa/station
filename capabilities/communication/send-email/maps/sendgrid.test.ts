import { sendEmailTest } from './send-email';

sendEmailTest('sendgrid', {
  from: 'hello@superface.ai',
  to: 'hello@superface.ai',
});
