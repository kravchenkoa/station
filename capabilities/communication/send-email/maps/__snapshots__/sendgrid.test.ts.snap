// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`communication/send-email/sendgrid SendEmail when all inputs are correct should return messageId as result 1`] = `
Object {
  "value": Object {
    "messageId": "lzQgRi0xSDGJzSXdcKjXFA",
  },
}
`;

exports[`communication/send-email/sendgrid SendEmail when inputs are invalid should throw an exception 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[4].responseHandlers[1].statements[0]
Original Map Location: Line 45, column 7",
}
`;
